capture/d2l-capture-producer/src/d2l-video-producer-captions.js

```diff
@@ -1,26 +1,351 @@
 				box-sizing: border-box;
 				position: relative;
 				width: 360px;
 				height: 532px;
 			}
-		`;
+
+			.d2l-video-producer-empty-captions-menu {
+				height: 100%;
+				display: flex;
+				flex-direction: column;
+				align-items: center;
+				justify-content: center;
+			}
+
+			.d2l-video-producer-captions-cues-list {
+				display: flex;
+				flex-direction: column;
+				height: 481px;
+				overflow-x: hidden;
+				overflow-y: scroll;
+				padding: 39px 10px 10px 10px;
+				position: relative;
+			}
+
+			.d2l-video-producer-captions-cues-list-bottom {
+				display: flex;
+				justify-content: center;
+				align-items: center;
+				width: 100%;
+				padding: 20px;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.captionCues = [];
+		this.visibleCuesInList = [];
+		this.intersectionObserver = null;
+
+		this._updateVisibleCuesInList = this._updateVisibleCuesInList.bind(this);
+	}
+
+	async firstUpdated() {
+		const fileUploader = this.shadowRoot.querySelector('#file-uploader');
+		fileUploader.addEventListener('d2l-file-uploader-files-added', this._onFilesAdded.bind(this));
 	}
 
 	render() {
 		return html`
 			<div class="d2l-video-producer-captions">
+				${this.captionCues.length > 0 ? this._renderCuesList() : this._renderEmptyCaptionsMenu()}
+				<d2l-alert-toast
+					id="d2l-video-producer-captions-alert-toast"
+				></d2l-alert-toast>
+			</div>
+			${this._appendVttScript()}
+		`;
+	}
+
+	updated(changedProperties) {
+		changedProperties.forEach((oldValue, propName) => {
+			if (propName === 'captionCues') {
+				if (oldValue && oldValue.length === 0 && this.captionCues.length > 0) {
+					const options = {
+						root: this.shadowRoot.querySelector('.d2l-video-producer-captions-cues-list'),
+						rootMargin: '0px',
+						threshold: 0.1,
+					};
+					const listBottom = this.shadowRoot.querySelector('.d2l-video-producer-captions-cues-list-bottom');
+					this.intersectionObserver = new IntersectionObserver(this._updateVisibleCuesInList, options);
+					this.intersectionObserver.observe(listBottom);
+				} else if (this.intersectionObserver && oldValue.length > 0 && this.captionCues.length === 0) {
+					this.intersectionObserver.disconnect();
+				}
+			}
+		});
+	}
+
+	// <script> tags must be added via Javascript in LitElement.
+	// https://stackoverflow.com/a/55693185
+	_appendVttScript() {
+		const script = document.createElement('script');
+		script.src = '../scripts/vtt.min.js';
+		return script;
+	}
+
+	_onFilesAdded(event) {
+		if (event && event.detail && event.detail.files.length === 1) {
+			const file = event.detail.files[0];
+			const extension = file.name.split('.').pop();
+			const alertToast = this.shadowRoot.querySelector('#d2l-video-producer-captions-alert-toast');
+			if (!['srt', 'vtt'].includes(extension.toLowerCase())) {
+				alertToast.setAttribute('open', 'open');
+				alertToast.setAttribute('type', 'critical');
+				alertToast.innerText = this.localize('captionsInvalidFileType');
```

Comment:
Perhaps instead of setting the `alertToast` here, we could add a helper to reuse, e.g.,
```js
_openAlertToast({type, text}) {
    const alertToast = this.shadowRoot.querySelector('...');
    alertToast.setAttribute('...');
}
```

And update it to be:
```suggestion
				this._openAlertToast({type: 'critical', text: this.localize('captionsInvalidFileType')});
```